packages/@ember/-internals/glimmer/lib/helpers/or.ts

```diff
@@ -22,7 +22,14 @@ import { InternalHelperReference } from '../utils/references';
   @since 2.7.0
 */
 function or({ positional: { references } }: CapturedArguments) {
-  return references.reduce((acc: any, ref) => acc || ref.value(), false);
+  let last: any = false;
+  for (let i = 0; i < references.length; i++) {
+    last = references[i].value();
+    if (last) {
+      return last;
+    }
+  }
+  return last;
```

Comment:
I don't know if V8 already does it, but one could try to remove the local variable to `return true` from within the loop or `return false` in the end

Reply:
```
for (let i = 0; i < references.length; i++) {
  if (references[i].value()) {
    return true;
  }
}
return false;
```

Reply:
I can't. And and or don't always return boolean, they return the last/first truthy value, whatever that is

Reply:
I see, makes sense